src/Handler/StreamHandler.php

```diff
@@ -118,8 +124,14 @@ private function createResponse(
             }
         }
 
-        if ($sink !== $stream) {
-            $this->drain($stream, $sink);
+        // Do not drain when the request is a HEAD request because they have
+        // no body.
+        if ($hasResponseBody && $sink !== $stream) {
```

Comment:
`$hasResponseBody` seems useless as the second condition is never true in this case anyway